File: 백준/Silver/10828. 스택/스택.js
const soluction = (inputs) => { 
    let stack = []
    let answer = ''
    let result = ''
    inputs.forEach((el) => { 
        switch (el[0]) { 
            case 'push':
                stack.push(el[1])
                break;
            case 'pop':
                result = stack.length !== 0 ? stack.pop() : -1
                answer += result + ' ';
                break;
            case 'size':
                result = stack.length
                answer += result + ' ';
                break;
            case 'empty':
                result = stack.length===0?1:0
                answer += result + ' ';
                break;
            case 'top':
                result = stack.length!==0?stack[stack.length -1]:-1
                answer += result + ' ';    
            default:
                break;
        }
    })
    console.log(answer.split(' ').join('\n'))
}

const readline = require("readline");
const rl = readline.createInterface({
	input: process.stdin,
	output: process.stdout,
});
let totalInputs = 0;
let inputs = [];

const insertInput = () => { 
    if (inputs.length < totalInputs) { 
        rl.on("line", (line) => { 
            inputs.push(line.split(' '));
        }).on("close", () => { 
            soluction(inputs);
            process.exit();
        })
    }
}

rl.question("", (count) => { 
    totalInputs = parseInt(count);
    insertInput();
})

File: 백준/Silver/10828. 스택/스택.py
def soluction(inputs):
    stack = []
    answer = ''

    for el in inputs:
        if el[0] == 'push':
            stack.append(el[1])
        elif el[0] == 'pop':
            result = stack.pop() if stack else -1
            answer += str(result) + '\n'
        elif el[0] == 'size':
            result = len(stack)
            answer += str(result) + '\n'
        elif el[0] == 'empty':
            result = 1 if not stack else 0
            answer += str(result) + '\n'
        elif el[0] == 'top':
            result = stack[-1] if stack else -1
            answer += str(result) + '\n'

    print(answer)
   
def main():
    total_count=int(input())
    inputs = []

    for _ in range(total_count):
        command = input().split()
        inputs.append(command)

    soluction(inputs)

if __name__ == "__main__":
    main()